rust/core-lib/src/view.rs

```diff
@@ -297,11 +307,13 @@ impl View {
         self.size = size;
     }
 
-    pub fn set_scroll(&mut self, first: i64, last: i64) {
+    pub fn set_scroll(&mut self, text: &Rope, first: i64, last: i64) {
         let first = max(first, 0) as usize;
         let last = max(last, 0) as usize;
         self.first_line = first;
         self.height = last - first;
+        let eof_line = self.line_of_offset(text, text.len()) + 1;
```

Comment:
Is +1 needed here?

Reply:
It is if this value is being used as either the "number of lines" or as the end bound of a half open range. That is to say: if a document has 'one' line, `line_of_offset(text.len()) == 0`.

Reply:
Ok. Then +1 here looks correct.